lib/cypress/demographics_randomizer.rb

```diff
@@ -126,5 +126,21 @@ def self.get_random_payer_start_date(patient)
         patient.birthDatetime
       end
     end
+
+    def self.randomize_birthdate(patient, random: Random.new)
+      birth_datetime = patient.birthDatetime
+      while birth_datetime == patient.birthDatetime
+        patient.birthDatetime = patient.birthDatetime.change(
+          case random.rand(3)
+          when 0 then { 'day' => 1, 'month' => 1 }
+          when 1 then { 'day' => random.rand(28) + 1 }
+          when 2 then { 'month' => random.rand(12) + 1 }
+          end
+        )
+      end
+      if patient.dataElements.where(_type: QDM::PatientCharacteristicBirthdate).first
```

Comment:
This works as is however gor consistency, in lib/ext/patient.rb the other patient characteristics are accessed using the pattern below.  If you like the pattern below, and it works for you, feel free to use it.

gender_chars = get_data_elements('patient_characteristic', 'gender')